utils/RunONNXModel.py

```diff
@@ -623,13 +625,33 @@ def main():
                 exit(1)
 
             # Verify using softmax first.
-            if (args.verify_with_softmax):
+            if (args.verify_with_softmax is not None):
                 for i, name in enumerate(output_names):
                     print(
-                        "Verifying using softmax for output {}:{}".format(name,
-                                                          list(outs[i].shape)))
-                    np.testing.assert_allclose(softmax(outs[i]), softmax(ref_outs[i]))
-                    print("  correct.\n")
+                        "Verifying using softmax along with"
+                        " axis {}".format(args.verify_with_softmax),
+                        " for output {}:{}".format(name, list(outs[i].shape)))
+                    total_elements = 0
+                    mismatched_elements = 0
+                    softmax_outs = softmax(outs[i])
+                    softmax_ref_outs = softmax(ref_outs[i])
+                    for index, actual_val in np.ndenumerate(softmax_outs):
+                        total_elements += 1
+                        ref_val = softmax_ref_outs[index]
+                        # Use equation atol + rtol * abs(desired), that is used in assert_allclose.
+                        diff = float(args.atol) + float(args.rtol) * abs(ref_val)
+                        if (abs(actual_val - ref_val) <= diff):
+                            continue
+                        mismatched_elements += 1
+                        print("  at {}".format(index),
+                              "mismatch {} (actual)".format(actual_val),
+                              "vs {} (reference)".format(ref_val))
+                    if mismatched_elements == 0:
+                        print("  correct.\n")
+                    else:
+                        raise AssertionError(
+                            "  mismatched elements {}/{}.\n".format(
+                                mismatched_elements, total_elements))
```

Comment:
Looks like the verification here is quite simlar to the one in `verify_every_value`, could you extract the common code into a function, then call the function? Thanks!

Reply:
Done. Thanks!